docs/design/metainfo-rpc.md

```diff
@@ -0,0 +1,147 @@
+# Metainfo RPC Service Refactor
+
+## Background
+
+Our current metainfo rpc service isn't very future proof.
+If we shipped our current client code, it wouldn't be easy to update clients and add support for the following:
+- versions
+- multipart uploads
+- cleanup for uncommitted segments
+- general concurrency fixes
+
+Native multipart upload can't happen with our current architecture because there's no concept of starting and ending a stream.
```

Comment:
The issue is in out of order uploading of a single stream not starting/ending a stream.